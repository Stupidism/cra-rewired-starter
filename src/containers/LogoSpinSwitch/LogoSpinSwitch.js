import React from 'react';
import { Switch } from 'antd';
import { compose, withHandlers } from 'recompose';

import logoSpinning from 'store/selectors/logoSpinning';
import connect from 'store/connect';
import { setRuntime } from 'store/modules/runtime';

export const LogoSpinSwitch = ({ logoSpinning, toggleLogoSpin }) => (
  <Switch
    checked={logoSpinning}
    onChange={toggleLogoSpin}
    checkedChildren="Toggle"
    unCheckedChildren="Toggle"
  />
);

export const toggleLogoSpin = ({ logoSpinning, dispatch }) => () =>
  dispatch(setRuntime({ logoSpinning: !logoSpinning }));

export default compose(
  connect({ logoSpinning }),
  withHandlers({ toggleLogoSpin }),
)(LogoSpinSwitch);
