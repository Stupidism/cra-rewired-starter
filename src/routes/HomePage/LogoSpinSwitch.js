import React from 'react';
import { Switch } from 'antd';
import { compose, withHandlers } from 'recompose';

import logoSpinning from 'store/selectors/logoSpinning';
import connect from 'store/connect';
import { setRuntime } from 'store/modules/runtime';

export const LogoSpinSwitch = ({ logoSpinning, toggleLogoSpin }) => (
  <Switch checked={logoSpinning} onChange={toggleLogoSpin} />
);

export const toggleLogoSpin = ({ logoSpinning, dispatch }) => () =>
  dispatch(setRuntime({ logoSpinning: !logoSpinning }));

export default compose(
  connect({ logoSpinning }),
  withHandlers({ toggleLogoSpin }),
)(LogoSpinSwitch);
