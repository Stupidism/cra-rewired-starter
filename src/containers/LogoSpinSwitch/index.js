import LogoSpinSwitch from './LogoSpinSwitch';

export default LogoSpinSwitch;
