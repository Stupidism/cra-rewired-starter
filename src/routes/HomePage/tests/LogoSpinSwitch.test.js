import React from 'react';
import { shallow } from 'enzyme';

import { LogoSpinSwitch, toggleLogoSpin } from '../LogoSpinSwitch';

describe('<LogoSpinSwitch />', () => {
  it('should be consistent with snapshot', () => {
    const snapshot = shallow(<LogoSpinSwitch logoSpinning />);
    expect(snapshot).toMatchSnapshot();
  });
});

describe('toggleLogoSpin({ logoSpinning, dispatch }) => handler', () => {
  it('should set runtime.logoSpinning inversely', function() {
    const dispatch = jest.fn();

    toggleLogoSpin({ logoSpinning: true, dispatch })();
    toggleLogoSpin({ logoSpinning: false, dispatch })();

    expect(dispatch.mock.calls).toMatchSnapshot();
  });
});
