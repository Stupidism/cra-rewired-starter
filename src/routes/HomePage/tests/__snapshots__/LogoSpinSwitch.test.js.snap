// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoSpinSwitch /> should be consistent with snapshot 1`] = `
<Switch
  checked={true}
  prefixCls="ant-switch"
/>
`;

exports[`toggleLogoSpin({ logoSpinning, dispatch }) => handler should set runtime.logoSpinning inversely 1`] = `
Array [
  Array [
    Object {
      "type": "SET_RUNTIME_VARIABLE",
      "value": Object {
        "logoSpinning": false,
      },
    },
  ],
  Array [
    Object {
      "type": "SET_RUNTIME_VARIABLE",
      "value": Object {
        "logoSpinning": true,
      },
    },
  ],
]
`;
